### ISSUE TITLE
Staging Review - Accessibility Feedback - converted Modal component

### ISSUE THREAD

**Issue by caw310** [department-of-veterans-affairs/vets-design-system-documentation#2087]
## Description
Feedback from Brian for Staging review for Modal
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

### Modal
**Must**: When a [modal doesn't have a title](https://design.va.gov/components/modal#without-title), the containing div has an aria-label="undefined modal" which then gets announced to screen reader users. Needs to be a more graceful way of handling this than saying "undefined" when the title prop is undefined.
Should: This might solve the "must" feedback --- does it ever make sense for a modal to not have a title? I feel like the [USWDS accessibility guidance](https://designsystem.digital.gov/components/modal/#accessibility-modal) kind of implies that modals should always have a title, though it doesn't say it explicitly. My preference would be for us to codify that in the design system, all modals need to have a title.
Weird bug? / 

**Must**: In Storybook, when you have the [Crisis Line modal](https://design.va.gov/storybook/?path=/docs/uswds-va-modal--default#crisis-line-modal) example open and tab through the elements inside the modal, the page behind the modal scrolls down. That unexpected motion could be disorienting, and might be confusing when you close the modal. But it's just on the one example, which makes me think it might be a weird bug with that modal in Storybook rather than a problem with the component.
**Consider**: There's been some movement toward using visible "CLOSE" text alongside the "X" button in other contexts, like the notification and card components. I'd love to see that happen here too!

## Details
This ticket is only to address the Must items. 

## Tasks
- [ ] Address the Must items identified by Brian
- [ ] Raise the 'consider' suggestion with the team

## Acceptance Criteria
- [ ] Modal has been updated with the issues identified during staging review.

**Comment by laflannery** [comment]
I would like to revisit the ask here because I believe there was some confusion. You are correct that a modal does always need to have a "title" but that does not mean that it must have the `title` property. What this requirement means is that there needs to be some element within the modal that is programmatically indicated as the title of the modal.

The standard, most common way to do this is to have a heading within the modal and then this heading is associated to the modal using an `aria-labelledby` attribute. I would like to point you to [a bug ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1407) I filed last year where I supported the Public Websites team in building the new homepage modal and we did this. The reason a bug ticket was filed was for the reason you have identified here - we had "undefined modal" in the `aria-label` attribute because we weren't using `modalTitle`. However, we don't need to, what we would need in this cause is for `aria-label` to be completely gone if `modalTitle` is not used.

**Comment by rsmithadhoc** [comment]
@powellkerry I discussed with Laura and here is what we would like to do to satisfy her concerns and the governance feedback:

* Make `modalTitle` optional.
* Add a condition that if `modalTitle` is not set or is set to an empty string, _do not_ include `aria-label="undefined modal"` or `aria-label=" modal"` at all.

Their team is setting a title through `aria-labeledby`, which is the proper way to do it and the way USWDS recommends. We should update our component to match but that's outside the scope of this, so I will create a separate issue for updates to bring this in line with USWDS.
